src/write_sensu.c

```diff
@@ -39,6 +39,53 @@
 #include <stddef.h>
 
 #include <stdlib.h>
+#ifdef HAVE_KERNEL_SOLARIS
```

Comment:
Ah, thanks. Now that I understand it, it should be easy to fix :)
I always thought that AC_CHECK_FUNCS only checked for function declarations, but now it's obvious that is also tries to link with it (which is a good thing).

As for the fix, isn't it enough to just add
```
#undef vasprintf
#undef asprintf
```
in the substitute section?